functorch/csrc/BatchRulesReduceOps.cpp

```diff
@@ -275,6 +275,101 @@ std::tuple<Tensor, optional<int64_t>, Tensor, optional<int64_t>> aminmax_batchin
   return std::make_tuple(min, 0, max, 0);
 }
 
+std::tuple<Tensor, Tensor> cdist_batch_rule_helper(
+    const Tensor& x1, optional<int64_t> x1_bdim,
+    const Tensor& x2, optional<int64_t> x2_bdim) {
+
+  auto x1_ = moveBatchDimToFront(x1, x1_bdim);
+  auto x2_ = moveBatchDimToFront(x2, x2_bdim);
+
+  int64_t d1 = static_cast<int64_t>(x1_bdim.has_value());
+  int64_t d2 = static_cast<int64_t>(x2_bdim.has_value());
+  while (x2_bdim && x1_.dim() - d1 > x2_.dim() - d2) {
+    // x1:                      [  S1, S2, .............., SK]
+    // x2: [B, D1, ..., DN]  -> [B, 1,  1, ..., 1, D1 ..., DN]
+    x2_ = x2_.unsqueeze(1);
+  }
+  while (x1_bdim && x2_.dim() - d2 > x1_.dim() - d1) {
+    // x2:                      [  S1, S2, .............., SK]
+    // x1: [B, D1, ..., DN]  -> [B, 1,  1, ..., 1, D1 ..., DN]
+    x1_ = x1_.unsqueeze(1);
+  }
```

Comment:
I'm confused -- what is going on here?

Reply:
Is this the same logic as https://github.com/pytorch/functorch/blob/b1d08d4669596a83e1d4e3fa72842bc224dff531/functorch/csrc/BatchRulesBinaryOps.cpp#L48-L53 ?

Reply:
Looks like `_binary_pointwise_batch_rule` + making contiguous `tensor_` and `other_` can replace all the code for `_cdist_forward`. Thanks a lot for pointing out `maybePadToLogicalRank`.